packages/shared/src/components/cards/share/ShareList.tsx

```diff
@@ -107,13 +109,14 @@ export const ShareList = forwardRef(function ShareList(
           onShare={onShare}
           post={post}
           imageProps={{
-            loading: 'lazy',
             alt: 'Post Cover image',
-            src: post.sharedPost.image,
             className: classNames(
               'mobileXXL:self-start',
               !isVideoType && 'mt-4',
             ),
+            loading: 'lazy',
+            ...(eagerLoadImage && HIGH_PRIORITY_IMAGE_PROPS),
```

Comment:
It's a bit weird to have loading here as well as in the HIGH_PRIORITY_IMAGE_PROPS no?

Reply:
Yes, not the best. 
I put because we have to choose if keep the `HIGH_PRIORITY_IMAGE_PROPS` and reuse the props or put the ternary for each property. Do you think is an anti-pattern?

I'm open for proposal if there are other options :)

Reply:
Well maybe not for each but:
```suggestion
            ...(eagerLoadImage ? HIGH_PRIORITY_IMAGE_PROPS : {loading: 'lazy'}),
```

Reply:
Its just overriding props, from my end what Luca did looks logical and we do it quite often in other places 👍